### PR TITLE
test: add metadata api UT

### PR DESCRIPTION
**Pull request by BarryTong65**
### Description

add metadata api UT to 83.2% 
```
go test ./modular/metadata -coverprofile=coverage.out
ok      github.com/bnb-chain/greenfield-storage-provider/modular/metadata       1.175s  coverage: 83.2% of statements
```

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 
* add metadata api UT

### CODE REVIEW

**Review by constwz — APPROVED** [review]

File: modular/metadata/metadata_sp_exit_service.go [hunk 1]
```diff
@@ -59,10 +59,6 @@ func (r *MetadataModular) GfSpGetGlobalVirtualGroupByGvgID(ctx context.Context,
 	}
 
 	if gvg != nil {
-		if err != nil {
-			log.CtxErrorw(ctx, "failed to parse global virtual group ids", "error", err)
-			return nil, err
-		}
```
Comment (annielz) [inline modular/metadata/metadata_sp_exit_service.go:65]: can we continue execution here with error not nil?

Comment (BarryTong65) [reply modular/metadata/metadata_sp_exit_service.go:65]: yep, this is just redundant code by mistake
image

**Review by annielz — APPROVED** [review]

File: modular/metadata/metadata_query_service_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	"go.uber.org/mock/gomock"
+	"gorm.io/gorm"
+
+	"github.com/bnb-chain/greenfield-storage-provider/modular/metadata/types"
+)
+
+func TestMetadataModular_GfSpQueryUploadProgress_Success(t *testing.T) {
+	a := setup(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
```
Comment (sysvm) [inline modular/metadata/metadata_query_service_test.go:20]: golang1.14+ doesn't need call `ctrl.Finish()`.

Comment (BarryTong65) [reply modular/metadata/metadata_query_service_test.go]: fixed

**Pull request merged**